### ISSUE TITLE
Issue with dynamic routes

### ISSUE THREAD

**Issue by jesperlandberg** [Dogstudio/highway#3]
Hi,

Thanks for this barba alternative=)

Currently testing it out but I'm having one issue. Links to Dynamic pages (Wordpress single pages) doesn't work properly. It works the first click/entry after a page refresh, but going between pages and then trying to enter a dynamic route causes a hard refresh. Any idea why?

JS
```
import '../css/app.scss'

import Highway from '@dogstudio/highway'

// Rendereres
import Home from './home'
import Location from './location'

(() => {
	const links = [...document.querySelectorAll('.js-site-head__menu-link')]

	const H = new Highway.Core({
		renderers: {
			home: Home,
			location: Location
		},
		transitions: {

		}
	})

	H.on('NAVIGATE_IN', (from, state) => {
		for (const link of links) {
		  link.classList.remove('is-active')

		  if (link.href === state.url) {
		    link.classList.add('is-active')
		  }
		}
	})

})();
```

Single renderer:
```
import Highway from '@dogstudio/highway'
import Smooth from '../scroll'

let smooth

class Location extends Highway.Renderer {
	onEnter() {
		smooth = new Smooth()

		smooth.init()
	}
	onLeave() {
		smooth.destroy()
	}
	onEnterCompleted() { }
	onLeaveCompleted() { }
}

export default Location
```

Single page file (twig templating)
```
{% extends "base.twig" %}
{% block content %}
<article router-view="location" class="o-scroll js-scroll">
	<div class="o-scroll__content js-scroll__content">

	</div>
</article>
{% endblock %}

```

**Comment by Anthodpnt** [comment]
@jesperlandberg 

I'm not sure to understand.
When you click a link to a single page from Wordpress the window reload happens ?
Do you have any error in the console ?

Thanks,
Anthodpnt

**Comment by Anthodpnt** [comment]
@jesperlandberg Do you have the `router-wrapper` as well in you HTML ?

**Comment by jesperlandberg** [comment]
@Anthodpnt the issue is with pages that have dynamic slugs, /:id for example. All pre defined pages works fine, but dynamic ones like blog posts and such seem to have issues. 

The issue for me (and a friend who is also trying out highway) seem to be that the second time we click a link to a blog post or dynamic page, a hard load of the pages happen.

"loads the site" -> clicks and enter blog post (works fine) -> goes back to home page -> clicks any blog post again (this time causing a hard load).

There are no errors show in the console or other logs.

I will try put it up on a test server for you to see later today.

**Comment by deJong** [comment]
In addition to @jesperlandberg, there is always a hard refresh of the page once a link is loaded dynamically in the router view and clicked.

The transition works, but after about 100ms there is still a hard refresh.

**Comment by Anthodpnt** [comment]
@jesperlandberg @deJong,

Thanks for your feedback!
I'll run some tests on my side and come back to you ASAP with a fix.

**Comment by jesperlandberg** [comment]
@deJong yeah, exactly the problem I tried to explain haha, you did it better=)

**Comment by Anthodpnt** [comment]
@jesperlandberg @deJong 

Ok I see, should be an issue with event binding.
Will check it and come back to you.

Thanks again !

**Comment by Anthodpnt** [comment]
@jesperlandberg @deJong 

Should be fixed in **[v1.3.4](https://github.com/Dogstudio/highway/releases/tag/v1.3.4)**.
The binding of the click event on links was triggered too soon...

Can you please update your package and tell me if this solves your issue please ?

Thanks a lot,
Anthodpnt

**Comment by jesperlandberg** [comment]
@Anthodpnt I'm afraid the issue still remains for me=/

**Comment by jesperlandberg** [comment]
@Anthodpnt My mistake. I ran a command to update my package.json, which displayed 1.3.3 -> 1.3.4 ... but  now I saw it didn't actually update my package file for some reason. 

However, it seems to work fine now=) Thanks for the quick update!
